server/plugin/api.go

```diff
@@ -690,32 +673,39 @@ func (p *Plugin) handleDeleteSubscriptions(w http.ResponseWriter, r *http.Reques
 		return
 	}
 
-	if statusCode, err := p.Client.DeleteSubscription(body.Organization, subscription.SubscriptionID, mattermostUserID); err != nil {
-		p.API.LogError(constants.DeleteSubscriptionError, "Error", err.Error())
-		p.handleError(w, r, &serializers.Error{Code: statusCode, Message: err.Error()})
-		return
+	statusCode, deleteErr := p.deleteSubscription(subscription, mattermostUserID)
+	if deleteErr != nil {
+		p.API.LogError(constants.DeleteSubscriptionError, "Error", deleteErr.Error())
+		p.handleError(w, r, &serializers.Error{Code: statusCode, Message: deleteErr.Error()})
+	}
+
+	returnStatusOK(w)
+}
+
+func (p *Plugin) deleteSubscription(subscription *serializers.SubscriptionDetails, mattermostUserID string) (int, error) {
+	if statusCode, err := p.Client.DeleteSubscription(subscription.OrganizationName, subscription.SubscriptionID, mattermostUserID); err != nil {
+		return statusCode, err
 	}
 
 	if deleteErr := p.Store.DeleteSubscription(&serializers.SubscriptionDetails{
-		MattermostUserID:             body.MMUserID,
-		ProjectName:                  body.Project,
-		OrganizationName:             body.Organization,
-		EventType:                    body.EventType,
-		ChannelID:                    body.ChannelID,
-		Repository:                   body.Repository,
-		TargetBranch:                 body.TargetBranch,
-		PullRequestCreatedBy:         body.PullRequestCreatedBy,
-		PullRequestReviewersContains: body.PullRequestReviewersContains,
-		PushedBy:                     body.PushedBy,
-		MergeResult:                  body.MergeResult,
-		NotificationType:             body.NotificationType,
-		AreaPath:                     body.AreaPath,
+		MattermostUserID:             subscription.MattermostUserID,
+		ProjectName:                  subscription.ProjectName,
+		OrganizationName:             subscription.OrganizationName,
+		EventType:                    subscription.EventType,
+		ChannelID:                    subscription.ChannelID,
+		Repository:                   subscription.Repository,
+		TargetBranch:                 subscription.TargetBranch,
+		PullRequestCreatedBy:         subscription.PullRequestCreatedBy,
+		PullRequestReviewersContains: subscription.PullRequestReviewersContains,
+		PushedBy:                     subscription.PushedBy,
+		MergeResult:                  subscription.MergeResult,
+		NotificationType:             subscription.NotificationType,
+		AreaPath:                     subscription.AreaPath,
 	}); deleteErr != nil {
-		p.API.LogError(constants.DeleteSubscriptionError, "Error", deleteErr.Error())
-		p.handleError(w, r, &serializers.Error{Code: http.StatusInternalServerError, Message: deleteErr.Error()})
+		return http.StatusInternalServerError, deleteErr
 	}
 
-	returnStatusOK(w)
+	return http.StatusOK, nil
```

Comment:
why this change?

Reply:
This is not a change we create a separate function. Github got little confused here